src/server/clients_info.rs

```diff
@@ -28,19 +34,45 @@ pub(crate) struct ClientBuffers {
 }
 
 impl ClientsInfo {
+    pub(super) fn new(policy: VisibilityPolicy) -> Self {
+        Self {
+            info: Default::default(),
+            policy,
+        }
+    }
+
+    /// Returns reference to connected client info.
```

Comment:
I think that we should rename it into `ClientState` and the parent struct into `ClientsState`.
But I think it's better to do as part of a separate PR.